src/System.Runtime.Extensions/src/System/Environment.Unix.cs

```diff
@@ -112,21 +114,209 @@ private static IDictionary GetEnvironmentVariablesCore(EnvironmentVariableTarget
                     break;
+            }
+
+            // No known path for the SpecialFolder
+            return string.Empty;
+        }
+
+        private static string GetXdgConfig(string home)
+        {
+            // "$XDG_CONFIG_HOME defines the base directory relative to which user specific configuration files should be stored."
+            // "If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used."
+            string config = GetEnvironmentVariable("XDG_CONFIG_HOME");
+            if (string.IsNullOrEmpty(config))
+            {
+                config = Path.Combine(home, ".config");
+            }
+            return config;
+        }
+
+        private static string ReadXdgDirectory(string homeDir, string key, string fallback)
+        {
+            Debug.Assert(!string.IsNullOrEmpty(homeDir), $"Expected non-empty homeDir");
+            Debug.Assert(!string.IsNullOrEmpty(key), $"Expected non-empty key");
+            Debug.Assert(!string.IsNullOrEmpty(fallback), $"Expected non-empty fallback");
 
-                // TODO: Add more special folder handling
+            string envPath = GetEnvironmentVariable(key);
+            if (!string.IsNullOrEmpty(envPath))
+            {
+                return envPath;
+            }
+
+            string userDirsPath = Path.Combine(GetXdgConfig(homeDir), "user-dirs.dirs");
+            if (Interop.Sys.Access(userDirsPath, Interop.Sys.AccessMode.R_OK) == 0)
+            {
+                try
+                {
+                    // TODO #11151: Replace with direct usage of File.ReadLines or equivalent once we have access to System.IO.FileSystem here.
+                    Type fileType = Type.GetType("System.IO.File, System.IO.FileSystem, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", throwOnError: false);
+                    IEnumerable<string> lines = fileType?.GetTypeInfo().GetDeclaredMethod("ReadLines").Invoke(null, new object[] { userDirsPath }) as IEnumerable<string>;
+                    if (lines != null)
+                    {
+                        foreach (string line in lines)
+                        {
+                            // Example line:
+                            // XDG_DESKTOP_DIR="$HOME/Desktop"
+
+                            // Skip past whitespace at beginning of line
+                            int pos = 0;
+                            while (pos < line.Length && char.IsWhiteSpace(line[pos])) pos++;
+                            if (pos >= line.Length) continue;
+
+                            // Skip past requested key name
+                            if (string.CompareOrdinal(line, pos, key, 0, key.Length) != 0) continue;
+                            pos += key.Length;
+
+                            // Skip until and past '='
+                            while (pos < line.Length && line[pos] != '=') pos++; // skip until '='
```

Comment:
Do you want to check that `line[pos]` is whitespace rather than not `=`? It seems this would e.g. match `XDG_DESKTOP_DIR_FOO="bar"` by ignoring the `_FOO` part when scanning for `=`.